### PR TITLE
ffi: expose the notification settings

### PR DESCRIPTION
**Pull request by nimau**
This PR is the last step of the https://github.com/matrix-org/matrix-rust-sdk/issues/1959 implementation.

It exposes the `notification_settings` crate to update notification settings for a room, and to define whether certain predefined push rules are enabled.

A `NotificationSettingsDelegate` can be defined to notify the client application each time push rules are updated.

### CODE REVIEW

**Review by poljar — CHANGES_REQUESTED** [review]
Looks good, the ULD needs a newline at the end and I believe that we can even skip modifying the UDL completely.

File: bindings/matrix-sdk-ffi/src/api.udl [hunk 1]
```diff
@@ -77,3 +77,7 @@ enum EventItemOrigin {
     "Sync",
     "Pagination",
 };
+
+callback interface NotificationSettingsDelegate {
+    void settings_did_change();
+};
```
[the remark this answers is not in the document]
Comment (jplatte) [reply bindings/matrix-sdk-ffi/src/api.udl:83]: We can :)

```rust
#[uniffi::export(callback_interface)]
trait NotificationSettingsDelegate { /* ... */ }
```

[the remark this answers is not in the document]
Comment (poljar) [reply bindings/matrix-sdk-ffi/src/api.udl:83]: Right, let's do that then. Thanks.

[the remark this answers is not in the document]
Comment (nimau) [reply bindings/matrix-sdk-ffi/src/api.udl]: Oh nice. Ok, I've updated the branch.

**Review by poljar — APPROVED** [review]
Looks good. If CI passes we can merge.

**Pull request merged**